src/instructlab/utils.py

```diff
@@ -564,3 +595,53 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     if not hostname or not port:
         raise ValueError(f"Invalid host-port string: '{hostport}'")
     return hostname, port
+
+
+def convert_messages_to_legacy_dataset(
+    dataset: List[MessageSample],
+) -> List[LegacyMessageSample]:
+    """
+    Converts the new HuggingFace messages dataset format to the legacy format.
+    **Note**: The legacy dataset format assumes only a turn of 1. All extra turns will be dropped.
+    """
+    converted_dataset: List[LegacyMessageSample] = []
+    for dp in dataset:
+        # in new dataset, the roles of a message will be determined.
+        if len(dp["messages"]) < 3:
+            raise ValueError(
+                "Dataset cannot be missing messages from roles. Please make sure that a message from each role is present."
+            )
+        system = dp["messages"][0]["content"]
+        user = dp["messages"][1]["content"]
+        assistant = dp["messages"][2]["content"]
```

Comment:
Is there a way to ensure `["messages"][0]` is always `system` and so on? Additionally, shouldn't we validate the presence of `["content"]` too?

Reply:
Yeah, I inspected the format of our dataset in a separate Jupyter notebook and found that it exclusively used this order. The original implementation I had didn't assume this order, but it's likelier that we'll get rid of the legacy linux train before we fundamentally restructure our dataset format.

Reply:
Btw I fixed this to instead use a dict comprehension, since the final object will be a dict anyway, the order won't matter and therefore we can just form the object by doing:

```py
sample = {m["role"]: m["content"] for m in dp["messages"]}
```